flink-libraries/flink-table/src/main/scala/org/apache/flink/table/functions/aggfunctions/AvgAggFunction.scala

```diff
@@ -28,8 +28,9 @@ import org.apache.flink.table.functions.{Accumulator, AggregateFunction}
   * @tparam T the type for the aggregation result
   */
 abstract class IntegralAvgAggFunction[T] extends AggregateFunction[T] {
+
   /** The initial accumulator for Integral Avg aggregate function */
-  class IntegralAvgAccumulator extends Accumulator {
+  class IntegralAvgAccumulator extends JTuple2[Long, Long] with Accumulator {
```

Comment:
If you are using a `JTuple` you may not add additional fields but should use the Tuple2 fields (`f0` and `f1`). 
Since the `TupleSerializer` is not aware of the added fields, `IntegralAvgAccumulator` won't be treated as Tuple2 by Flink.